File: src/RealtimeServer/scriptureforge/models/sf-project-role.ts
export function canParatextRoleWrite(role: string | undefined): boolean {
  switch (role) {
    case SFProjectRole.ParatextAdministrator:
    case SFProjectRole.ParatextTranslator:
    case SFProjectRole.ParatextConsultant:
      return true;
    default:
      return false;
  }
}

export function isParatextRole(role: string | undefined): boolean {
  switch (role) {
    case SFProjectRole.ParatextAdministrator:
    case SFProjectRole.ParatextTranslator:
    case SFProjectRole.ParatextConsultant:
    case SFProjectRole.ParatextObserver:
      return true;
    default:
      return false;
  }
}

export function isTranslateRole(role: string | undefined): boolean {
  if (isParatextRole(role)) {
    return true;
  }
  switch (role) {
    case SFProjectRole.Commenter:
    case SFProjectRole.Viewer:
      return true;
    default:
      return false;
  }
}

export enum SFProjectRole {
  ParatextAdministrator = 'pt_administrator',
  ParatextTranslator = 'pt_translator',
  ParatextConsultant = 'pt_consultant',
  ParatextObserver = 'pt_observer',
  CommunityChecker = 'sf_community_checker',
  Commenter = 'sf_commenter',
  Viewer = 'sf_observer',
  None = 'none'
}

File: src/SIL.XForge.Scripture/Models/SFProjectRole.cs
namespace SIL.XForge.Scripture.Models;

public static class SFProjectRole
{
    public const string Administrator = "pt_administrator";
    public const string Translator = "pt_translator";
    public const string Consultant = "pt_consultant";
    public const string PTObserver = "pt_observer";
    public const string Commenter = "sf_commenter";
    public const string CommunityChecker = "sf_community_checker";
    public const string Viewer = "sf_observer";

    public static bool IsParatextRole(string role) =>
        role switch
        {
            Administrator or Translator or Consultant or PTObserver => true,
            _ => false,
        };
}
